src/probnum/linops/_linear_operator.py

```diff
@@ -1,11 +1,19 @@
 """Finite-dimensional linear operators."""
 
 import warnings
+from typing import Optional, Union
 
 import numpy as np
 import scipy.sparse.linalg
 import scipy.sparse.linalg.interface
 
+import probnum.utils
+from probnum.type import DTypeArgType, ScalarArgType, ShapeArgType
+
+BinaryOperandType = Union[
```

Comment:
Would it make sense to have this under `probnum.type`?

Reply:
Personally, I would opt for a little more locality here. To me it would seem a little out of place in `probnum.type`